### DISCUSSION TITLE
Problems in PAG reconstruction in multiphase microstructure

### DISCUSSION THREAD

**Discussion by atreyavibhor** [mtex-toolbox/mtex#698] [Ask Anything] [answered]
I am trying to reconstruct prior austenite grains from martensite in a microstructure which also contains ferrite (bcc) and very low retained austenite (fcc). I am using the desciption given at https://mtex-toolbox.github.io/MaParentGrainReconstructionAdvanced.html, which worked fine for reconstruction from fully martensite microstructure, but in current case is giving prolems. 

I have a dual phase (ferrite-martensite) microstructure like this:
![image](https://user-images.githubusercontent.com/52647372/104469039-70381880-55b8-11eb-8a7b-5433c7133871.png)

I identified the martensite based on grain average image quality (which I defined myself), switched off the ferrite part by equating phase=0, and ran the calcGrains function on martensite. The resulting orientation map looks like this (which is cool!):
![image](https://user-images.githubusercontent.com/52647372/104469806-361b4680-55b9-11eb-82d3-82351c4c18bb.png)

Now when I try to merge neighboring martensite grains to form austenite, using the merge function, that is where I see wierd result:
![image](https://user-images.githubusercontent.com/52647372/104470072-90b4a280-55b9-11eb-91bc-420cd799ee1e.png)
 
Basically calcGrains function is treating all the empty space as one big martensite grain, and then the merge function basically merges the other martensite grains into it to result in a huge austenite parent grain. Here is the change I did to the code, rest is same as given on the website.:
![image](https://user-images.githubusercontent.com/52647372/104472921-be4f1b00-55bc-11eb-9bcb-cb5c422ec6ae.png)

I am looking for a way so that calcGrains function somehow disregards those empty spaces left by ferrite (for which I assigned phase=0). Or may be you have some other way to solve this problem. Will be glad to receive any help!

**Comment by nyyssont** [comment, marked as answer]
Would you consider trying MTEX 5.6.0 and using the job class for the reconstruction?

Instructions for job class here:
https://mtex-toolbox.github.io/MaParentGrainReconstruction.html

Basically, after your last lines there you'd enter
```
job = parentGrainReconstructor(ebsd,grains);
```

And proceed from there as in the instructions on the job class help page. Any luck using this route?

-Tuomo

**Reply by atreyavibhor** [reply]
Yes! Works perfectly now. I was facing problems using the earlier MTEX version. Thanks for your help! :-)

Regards
Vibhor

**Comment by ghost** [comment]


**Reply by atreyavibhor** [reply]
Thanks for your reply!. Its resolved now, after I used the job class for reconstruction on MTEX 5.6. Works better for me without using the option 'unitCell'.

Regards
Vibhor
